python/instrumentation/openinference-instrumentation-llama-index/src/openinference/instrumentation/llama_index/_callback.py

```diff
@@ -234,14 +235,24 @@ def on_event_start(
                 attributes = {}
 
         start_time = time_ns()
-        span: trace_api.Span = self._tracer.start_span(name=event_type.value, start_time=start_time)
+        context = None
+        if parent_id != BASE_TRACE_EVENT:
+            if parent_event_data := self._event_data.get(parent_id):
+                context = parent_event_data.context
+        span: trace_api.Span = self._tracer.start_span(
+            name=event_type.value,
+            start_time=start_time,
+            context=context,
+        )
```

Comment:
Nit: might want to put a little bit of docstrings here - it's a bit non-self evident.